cyclonedx/model/bom.py

```diff
@@ -644,23 +644,36 @@ def has_vulnerabilities(self) -> bool:
         """
         return bool(self.vulnerabilities)
 
-    def register_dependency(self, target: Dependable, depends_on: Optional[Iterable[Dependable]] = None) -> None:
+    def register_dependency(
+        self,
+        target: Dependable,
+        depends_on: Optional[Iterable[Dependable]] = None,
+        provides: Optional[Iterable[Dependable]] = None,
```

Comment:
instead of adding a new parameter here, how about adding a new method instead: `register_provision(self, target: Dependable, provides: Optional[Iterable[Dependable]] = None)`.


what do you think about this? 
this would fit the original architectural plans better.

Reply:
I'll give this a try